pipeline/stop_places/sql/stop_places.sql

```diff
@@ -531,6 +531,51 @@ $$
 LANGUAGE SQL IMMUTABLE STRICT;
 
 
+/*
+ * Create a function, that converts a duration string to the xsd:duration format.
+ * Returns null when no duration can be parsed
+ */
+CREATE OR REPLACE FUNCTION duration_to_xsd_duration(duration text) RETURNS text AS
+$$
+DECLARE
+  duration_epoch INTEGER;
+BEGIN
+  duration_epoch := EXTRACT(epoch FROM duration::INTERVAL)::INTEGER;
+  IF duration_epoch IS NULL THEN
+    RETURN NULL;
+  ELSEIF duration_epoch = 0 THEN
+    RETURN 'PT0S';
+  ELSE
+    RETURN concat('PT', (duration_epoch/3600)::INTEGER, 'H', ((duration_epoch%3600)/60)::INTEGER, 'M', duration_epoch%60, 'S');
+  END IF;
+END
+$$
+LANGUAGE plpgsql IMMUTABLE STRICT;
```

Comment:
I just re-looked this up and it seems like there is a better approach to deal with durations in postgresql.

We can use
`SELECT interval '33:22:11';` - interpreted correctly as `PT33H22M11S`
`SELECT interval '33:22';` - interpreted correctly as `PT33H22M`
`SELECT interval '33';` << unfortunately this is interpreted as seconds (`PT33S`), while it is supposed to describe minutes (https://wiki.openstreetmap.org/wiki/Key:interval)

https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-interval/

So we can change the function to simply do:

```
SET LOCAL intervalstyle = 'iso_8601';
SELECT interval duration;
```

Unfortunately the function additionally needs to check whether the given text is just a single number and then append "minutes" or ".00" to the string. E.g. `SELECT interval '33 minutes';`